src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -614,6 +618,13 @@ public Instance instance() throws IOException {
     instance.setName(uniqueName());
     instance.setDescription(description);
     instance.setZone(nameFromSelfLink(zone));
+    instance.setMetadata(newMetadata());
```

Comment:
Templates can have own metadata..
We dont want to overwrite them..

We should set it only when there is no template..
And if there is then we should only "add" ssh key.. as we are doing with labels..

Reply:
Modified to include the metadata from the template.

Reply:
cool..